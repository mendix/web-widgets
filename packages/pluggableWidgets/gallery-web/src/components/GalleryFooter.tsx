import classNames from "classnames";
import { createElement, ReactElement } from "react";

type GalleryFooterProps = Omit<JSX.IntrinsicElements["div"], "ref">;

export function GalleryFooter({ children, className, ...rest }: GalleryFooterProps): ReactElement {
    return (
        <div className={classNames("widget-gallery-footer", className)} {...rest}>
            {children}
        </div>
    );
}
