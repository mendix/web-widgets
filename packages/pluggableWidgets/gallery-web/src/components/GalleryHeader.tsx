import { createElement, ReactElement } from "react";

type GalleryHeaderProps = Omit<JSX.IntrinsicElements["div"], "ref">;

export function GalleryHeader(props: GalleryHeaderProps): ReactElement | null {
    const { children } = props;

    if (!children) {
        return null;
    }

    return <section {...props} className="widget-gallery-header widget-gallery-filter" />;
}
