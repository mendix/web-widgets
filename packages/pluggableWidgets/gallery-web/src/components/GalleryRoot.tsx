import classNames from "classnames";
import { ReactElement, createElement } from "react";

export type GalleryRootProps = Omit<JSX.IntrinsicElements["div"], "ref"> & {
    selectable?: boolean;
};

export function GalleryRoot(props: GalleryRootProps): ReactElement {
    const { className, selectable, children, ...rest } = props;

    return (
        <div
            className={classNames(
                "widget-gallery",
                {
                    "widget-gallery-selectable": selectable
                },
                className
            )}
            {...rest}
        >
            {children}
        </div>
    );
}
