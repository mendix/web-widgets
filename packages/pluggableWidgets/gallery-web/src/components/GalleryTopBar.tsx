import { createElement, ReactElement, PropsWithChildren } from "react";

export function GalleryTopBar(props: PropsWithChildren): ReactElement {
    return <div className="widget-gallery-top-bar">{props.children}</div>;
}
