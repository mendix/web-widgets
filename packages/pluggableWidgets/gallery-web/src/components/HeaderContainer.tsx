import { createElement } from "react";
import { getGlobalFilterContextObject } from "@mendix/widget-plugin-filtering/context";
import { getGlobalSortContext } from "@mendix/widget-plugin-sorting/context";
import { SortAPIProvider } from "@mendix/widget-plugin-sorting/providers/SortAPIProvider";
import { HeaderFiltersStore } from "@mendix/widget-plugin-filtering/stores/generic/HeaderFiltersStore";
import {
    getGlobalSelectionContext,
    SelectionHelper,
    useCreateSelectionContextValue
} from "@mendix/widget-plugin-grid/selection";
const SelectionContext = getGlobalSelectionContext();
const FilterContext = getGlobalFilterContextObject();
const SortAPIContext = getGlobalSortContext();

function FilterAPIProvider(props: React.PropsWithChildren<{ filtersStore: HeaderFiltersStore }>): React.ReactElement {
    return <FilterContext.Provider value={props.filtersStore.context}>{props.children}</FilterContext.Provider>;
}

function SelectionStatusProvider(
    props: React.PropsWithChildren<{ selectionHelper?: SelectionHelper }>
): React.ReactElement {
    const value = useCreateSelectionContextValue(props.selectionHelper);
    return <SelectionContext.Provider value={value}>{props.children}</SelectionContext.Provider>;
}

interface HeaderContainerProps
    extends React.PropsWithChildren<{
        filtersStore: HeaderFiltersStore;
        sortProvider: SortAPIProvider;
        selectionHelper?: SelectionHelper;
    }> {}

export function HeaderContainer(props: HeaderContainerProps): React.ReactElement {
    return (
        <FilterAPIProvider filtersStore={props.filtersStore}>
            <SortAPIContext.Provider value={props.sortProvider.context}>
                <SelectionStatusProvider selectionHelper={props.selectionHelper}>
                    {props.children}
                </SelectionStatusProvider>
            </SortAPIContext.Provider>
        </FilterAPIProvider>
    );
}
