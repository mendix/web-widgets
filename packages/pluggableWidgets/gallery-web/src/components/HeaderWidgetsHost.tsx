import { createElement } from "react";
// import { getGlobalSortContext } from "@mendix/widget-plugin-sorting/context";
import { getGlobalSelectionContext, useCreateSelectionContextValue } from "@mendix/widget-plugin-grid/selection";
import { useGalleryRootScope } from "../helpers/root-context";
const SelectionContext = getGlobalSelectionContext();
// const SortAPIContext = getGlobalSortContext();

export function HeaderWidgetsHost(props: { children?: React.ReactNode }): React.ReactElement {
    const { selectionHelper } = useGalleryRootScope();
    const selectionContext = useCreateSelectionContextValue(selectionHelper);

    return <SelectionContext.Provider value={selectionContext}>{props.children}</SelectionContext.Provider>;
}
