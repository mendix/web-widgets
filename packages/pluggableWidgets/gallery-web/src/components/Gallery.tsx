import { Pagination } from "@mendix/widget-plugin-grid/components/Pagination";
import { KeyNavProvider } from "@mendix/widget-plugin-grid/keyboard-navigation/context";
import { FocusTargetController } from "@mendix/widget-plugin-grid/keyboard-navigation/FocusTargetController";
import { PositionInGrid, SelectActionHandler } from "@mendix/widget-plugin-grid/selection";
import { ObjectItem } from "mendix";
import { createElement, ReactElement, ReactNode } from "react";
import { GalleryItemHelper } from "../typings/GalleryItem";
import { ListBox } from "./ListBox";
import { ListItem } from "./ListItem";
import { GalleryContent } from "./GalleryContent";
import { GalleryFooter } from "./GalleryFooter";
import { GalleryHeader } from "./GalleryHeader";
import { GalleryRoot } from "./GalleryRoot";
import { GalleryTopBar } from "./GalleryTopBar";

import { ItemEventsController } from "../typings/ItemEventsController";

export interface GalleryProps<T extends ObjectItem> {
    className?: string;
    desktopItems: number;
    emptyPlaceholderRenderer?: (renderWrapper: (children: ReactNode) => ReactElement) => ReactElement;
    emptyMessageTitle?: string;
    header?: ReactNode;
    headerTitle?: string;
    showHeader: boolean;
    hasMoreItems: boolean;
    items: T[];
    numberOfItems?: number;
    paging: boolean;
    page: number;
    pageSize: number;
    paginationPosition?: "below" | "above";
    showEmptyStatePreview?: boolean;
    phoneItems: number;
    setPage?: (computePage: (prevPage: number) => number) => void;
    tabletItems: number;
    tabIndex?: number;
    ariaLabelListBox?: string;
    preview?: boolean;

    // Helpers
    focusController: FocusTargetController;
    itemEventsController: ItemEventsController;
    itemHelper: GalleryItemHelper;
    selectHelper: SelectActionHandler;
    getPosition: (index: number) => PositionInGrid;
}

export function Gallery<T extends ObjectItem>(props: GalleryProps<T>): ReactElement {
    const pagination = props.paging ? (
        <div className="widget-gallery-pagination">
            <Pagination
                canNextPage={props.hasMoreItems}
                canPreviousPage={props.page !== 0}
                gotoPage={(page: number) => props.setPage && props.setPage(() => page)}
                nextPage={() => props.setPage && props.setPage(prev => prev + 1)}
                numberOfItems={props.numberOfItems}
                page={props.page}
                pageSize={props.pageSize}
                previousPage={() => props.setPage && props.setPage(prev => prev - 1)}
            />
        </div>
    ) : null;

    const showTopBar = props.paging && props.paginationPosition === "above";
    const showFooter = props.paging && props.paginationPosition === "below";

    return (
        <GalleryRoot className={props.className} selectable={false} data-focusindex={props.tabIndex || 0}>
            {showTopBar && <GalleryTopBar>{pagination}</GalleryTopBar>}
            {props.showHeader && <GalleryHeader aria-label={props.headerTitle}>{props.header}</GalleryHeader>}
            <GalleryContent hasMoreItems={props.hasMoreItems} setPage={props.setPage} isInfinite={!props.paging}>
                {props.items.length > 0 && (
                    <ListBox
                        lg={props.desktopItems}
                        md={props.tabletItems}
                        sm={props.phoneItems}
                        selectionType={props.selectHelper.selectionType}
                        aria-label={props.ariaLabelListBox}
                    >
                        <KeyNavProvider focusController={props.focusController}>
                            {props.items.map((item, index) => (
                                <ListItem
                                    preview={props.preview}
                                    key={`item_${item.id}`}
                                    helper={props.itemHelper}
                                    item={item}
                                    selectHelper={props.selectHelper}
                                    eventsController={props.itemEventsController}
                                    getPosition={props.getPosition}
                                    itemIndex={index}
                                />
                            ))}
                        </KeyNavProvider>
                    </ListBox>
                )}
            </GalleryContent>
            {(props.items.length === 0 || props.showEmptyStatePreview) &&
                props.emptyPlaceholderRenderer &&
                props.emptyPlaceholderRenderer(children => (
                    <section className="widget-gallery-empty" aria-label={props.emptyMessageTitle}>
                        <div className="empty-placeholder">{children}</div>
                    </section>
                ))}
            {showFooter && <GalleryFooter>{pagination}</GalleryFooter>}
        </GalleryRoot>
    );
}
