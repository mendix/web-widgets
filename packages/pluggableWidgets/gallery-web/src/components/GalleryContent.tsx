import { InfiniteBodyProps, useInfiniteControl } from "@mendix/widget-plugin-grid/components/InfiniteBody";
import classNames from "classnames";
import { ReactElement, ReactNode, createElement } from "react";

type PickProps = "hasMoreItems" | "setPage" | "isInfinite";

export type GalleryContentProps = {
    className?: string;
    children?: ReactNode;
} & Pick<InfiniteBodyProps, PickProps>;

export function GalleryContent({
    children,
    className,
    hasMoreItems,
    isInfinite,
    setPage
}: GalleryContentProps): ReactElement {
    const [trackScrolling, bodySize, containerRef] = useInfiniteControl({
        hasMoreItems,
        isInfinite,
        setPage
    });

    return (
        <div
            className={classNames("widget-gallery-content", { "infinite-loading": isInfinite }, className)}
            ref={containerRef}
            onScroll={isInfinite ? trackScrolling : undefined}
            style={isInfinite && bodySize > 0 ? { maxHeight: bodySize } : undefined}
        >
            {children}
        </div>
    );
}
