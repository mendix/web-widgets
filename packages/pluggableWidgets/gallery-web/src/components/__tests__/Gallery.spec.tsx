import { createElement } from "react";
import { Gallery } from "../../components/Gallery";
import { listAction, listExp } from "@mendix/widget-plugin-test-utils";
import "./__mocks__/intersectionObserverMock";
import "@testing-library/jest-dom";
import { waitFor, render } from "@testing-library/react";
import { mockProps, mockItemHelperWithAction, setup } from "../../utils/test-utils";
import { ItemHelperBuilder } from "../../utils/builders/ItemHelperBuilder";

describe("Gallery", () => {
    describe("DOM Structure", () => {
        it("renders correctly", () => {
            const { asFragment } = render(<Gallery {...mockProps()} />);

            expect(asFragment()).toMatchSnapshot();
        });

        it("renders correctly with onclick event", () => {
            const { asFragment } = render(
                <Gallery {...mockProps()} itemHelper={mockItemHelperWithAction(jest.fn())} />
            );

            expect(asFragment()).toMatchSnapshot();
        });
    });

    describe("with on click action", () => {
        it("runs action on item click", async () => {
            const execute = jest.fn();
            const props = mockProps({ onClick: listAction(mock => ({ ...mock(), execute })) });
            const { user, getAllByRole } = setup(<Gallery {...props} />);
            const [item] = getAllByRole("listitem");

            await user.click(item);
            await waitFor(() => expect(execute).toHaveBeenCalledTimes(1));
        });

        it("runs action on Enter|Space press when item is in focus", async () => {
            const execute = jest.fn();
            const props = mockProps({ onClick: listAction(mock => ({ ...mock(), execute })) });
            const { user, getAllByRole } = setup(<Gallery {...props} header={<span />} />);
            const [item] = getAllByRole("listitem");

            await user.tab();
            expect(item).toHaveFocus();
            await user.keyboard("[Enter]");
            await waitFor(() => expect(execute).toHaveBeenCalledTimes(1));
            await user.keyboard("[Space]");
            await waitFor(() => expect(execute).toHaveBeenCalledTimes(2));
        });
    });

    describe("with different configurations per platform", () => {
        it("contains correct classes for desktop", () => {
            const { getByRole } = render(<Gallery {...mockProps()} desktopItems={12} />);
            const list = getByRole("list");
            expect(list).toHaveClass("widget-gallery-lg-12");
        });

        it("contains correct classes for tablet", () => {
            const { getByRole } = render(<Gallery {...mockProps()} tabletItems={6} />);
            const list = getByRole("list");
            expect(list).toHaveClass("widget-gallery-md-6");
        });

        it("contains correct classes for phone", () => {
            const { getByRole } = render(<Gallery {...mockProps()} phoneItems={3} />);
            const list = getByRole("list");
            expect(list).toHaveClass("widget-gallery-sm-3");
        });
    });

    describe("with custom classes", () => {
        it("contains correct classes in the wrapper", () => {
            const { container } = render(<Gallery {...mockProps()} className="custom-class" />);

            expect(container.querySelector(".custom-class")).toBeVisible();
        });

        it("contains correct classes in the items", () => {
            const { getAllByRole } = render(
                <Gallery
                    {...mockProps()}
                    itemHelper={ItemHelperBuilder.sample(b => b.withItemClass(listExp(() => "custom-class")))}
                />
            );
            const [item] = getAllByRole("listitem");

            expect(item).toHaveClass("custom-class");
        });
    });

    describe("with pagination", () => {
        it("renders correctly", () => {
            const { asFragment } = render(
                <Gallery {...mockProps()} paging paginationPosition="above" numberOfItems={20} hasMoreItems />
            );

            expect(asFragment()).toMatchSnapshot();
        });

        it("triggers correct events on click next button", async () => {
            const setPage = jest.fn();
            const { user, getByLabelText } = setup(
                <Gallery
                    {...mockProps()}
                    paging
                    paginationPosition="above"
                    numberOfItems={20}
                    hasMoreItems
                    setPage={setPage}
                />
            );

            const next = getByLabelText("Go to next page");
            await user.click(next);
            await waitFor(() => expect(setPage).toHaveBeenCalledTimes(1));
        });
    });

    describe("with empty option", () => {
        it("renders correctly", () => {
            const { asFragment } = render(
                <Gallery
                    {...mockProps()}
                    items={[]}
                    emptyPlaceholderRenderer={renderWrapper => renderWrapper(<span>No items found</span>)}
                />
            );

            expect(asFragment()).toMatchSnapshot();
        });
    });

    describe("with accessibility properties", () => {
        it("renders correctly", () => {
            const { asFragment } = render(
                <Gallery
                    {...mockProps()}
                    items={[]}
                    headerTitle="filter title"
                    emptyMessageTitle="empty message"
                    emptyPlaceholderRenderer={renderWrapper => renderWrapper(<span>No items found</span>)}
                />
            );

            expect(asFragment()).toMatchSnapshot();
        });
    });

    describe("without filters", () => {
        it("renders structure without header container", () => {
            const props = { ...mockProps(), showHeader: false, header: undefined };
            const { asFragment } = render(<Gallery {...props} />);

            expect(asFragment()).toMatchSnapshot();
        });
    });
});
