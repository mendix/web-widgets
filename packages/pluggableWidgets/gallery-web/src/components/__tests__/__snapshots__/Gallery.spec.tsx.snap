// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery DOM Structure renders correctly 1`] = `
<DocumentFragment>
  <div
    class="widget-gallery my-gallery"
    data-focusindex="0"
  >
    <section
      aria-label="Mock props header aria label"
      class="widget-gallery-header widget-gallery-filter"
    >
      <input />
    </section>
    <div
      class="widget-gallery-content infinite-loading"
    >
      <div
        aria-label="Mock props ListBox aria label"
        class="widget-gallery-items widget-gallery-lg-4 widget-gallery-md-3 widget-gallery-sm-2"
        role="list"
      >
        <div
          class="widget-gallery-item item-class"
          data-position="0,0"
          role="listitem"
          tabindex="0"
        >
          Item content
        </div>
        <div
          class="widget-gallery-item item-class"
          data-position="1,0"
          role="listitem"
          tabindex="-1"
        >
          Item content
        </div>
        <div
          class="widget-gallery-item item-class"
          data-position="2,0"
          role="listitem"
          tabindex="-1"
        >
          Item content
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Gallery DOM Structure renders correctly with onclick event 1`] = `
<DocumentFragment>
  <div
    class="widget-gallery my-gallery"
    data-focusindex="0"
  >
    <section
      aria-label="Mock props header aria label"
      class="widget-gallery-header widget-gallery-filter"
    >
      <input />
    </section>
    <div
      class="widget-gallery-content infinite-loading"
    >
      <div
        aria-label="Mock props ListBox aria label"
        class="widget-gallery-items widget-gallery-lg-4 widget-gallery-md-3 widget-gallery-sm-2"
        role="list"
      >
        <div
          class="widget-gallery-item widget-gallery-clickable item-class"
          data-position="0,0"
          role="listitem"
          tabindex="0"
        >
          <div
            class="widget-gallery-item-button"
            role="button"
          >
            Item content
          </div>
        </div>
        <div
          class="widget-gallery-item widget-gallery-clickable item-class"
          data-position="1,0"
          role="listitem"
          tabindex="-1"
        >
          <div
            class="widget-gallery-item-button"
            role="button"
          >
            Item content
          </div>
        </div>
        <div
          class="widget-gallery-item widget-gallery-clickable item-class"
          data-position="2,0"
          role="listitem"
          tabindex="-1"
        >
          <div
            class="widget-gallery-item-button"
            role="button"
          >
            Item content
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Gallery with accessibility properties renders correctly 1`] = `
<DocumentFragment>
  <div
    class="widget-gallery my-gallery"
    data-focusindex="0"
  >
    <section
      aria-label="filter title"
      class="widget-gallery-header widget-gallery-filter"
    >
      <input />
    </section>
    <div
      class="widget-gallery-content infinite-loading"
    />
    <section
      aria-label="empty message"
      class="widget-gallery-empty"
    >
      <div
        class="empty-placeholder"
      >
        <span>
          No items found
        </span>
      </div>
    </section>
  </div>
</DocumentFragment>
`;

exports[`Gallery with empty option renders correctly 1`] = `
<DocumentFragment>
  <div
    class="widget-gallery my-gallery"
    data-focusindex="0"
  >
    <section
      aria-label="Mock props header aria label"
      class="widget-gallery-header widget-gallery-filter"
    >
      <input />
    </section>
    <div
      class="widget-gallery-content infinite-loading"
    />
    <section
      class="widget-gallery-empty"
    >
      <div
        class="empty-placeholder"
      >
        <span>
          No items found
        </span>
      </div>
    </section>
  </div>
</DocumentFragment>
`;

exports[`Gallery with pagination renders correctly 1`] = `
<DocumentFragment>
  <div
    class="widget-gallery my-gallery"
    data-focusindex="0"
  >
    <div
      class="widget-gallery-top-bar"
    >
      <div
        class="widget-gallery-pagination"
      >
        <div
          aria-label="Pagination"
          class="pagination-bar"
        >
          <button
            aria-label="Go to first page"
            class="btn pagination-button"
            disabled=""
          >
            <span
              aria-hidden="true"
            >
              <svg
                class="pagination-icon step-backward"
                fill="currentColor"
                viewBox="0 0 32 32"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M23.19 6.63L11.51 15.61C11.25 15.81 11.25 16.2 11.51 16.4L23.2 25.38C23.53 25.64 24.01 25.4 24.01 24.99V7.02C24.01 6.6 23.53 6.37 23.2 6.63H23.19Z"
                  fill="currentColor"
                />
                <path
                  d="M10 6H8V26H10V6Z"
                  fill="currentColor"
                />
              </svg>
            </span>
          </button>
          <button
            aria-label="Go to previous page"
            class="btn pagination-button"
            disabled=""
          >
            <span
              aria-hidden="true"
            >
              <svg
                class="pagination-icon backward"
                fill="currentColor"
                viewBox="0 0 32 32"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M14.19 6.63L3.51 15.61C3.25 15.81 3.25 16.2 3.51 16.4L14.2 25.38C14.53 25.64 15.01 25.4 15.01 24.99V7.02C15.01 6.6 14.53 6.37 14.2 6.63H14.19Z"
                  fill="currentColor"
                />
                <path
                  d="M27.19 6.63L16.5 15.61C16.24 15.81 16.24 16.2 16.5 16.4L27.19 25.38C27.52 25.64 28 25.4 28 24.99V7.02C28 6.6 27.52 6.37 27.19 6.63Z"
                  fill="currentColor"
                />
              </svg>
            </span>
          </button>
          <span
            class="sr-only sr-only-focusable"
          >
            Currently showing 1 to 10 of 20
          </span>
          <div
            aria-hidden="true"
            class="paging-status"
          >
            1 to 10 of 20
          </div>
          <button
            aria-label="Go to next page"
            class="btn pagination-button"
          >
            <span
              aria-hidden="true"
            >
              <svg
                class="pagination-icon forward"
                fill="currentColor"
                viewBox="0 0 32 32"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M17.81 6.63C17.48 6.37 17 6.61 17 7.02V24.98C17 25.4 17.48 25.63 17.81 25.37L28.5 16.39C28.76 16.19 28.76 15.8 28.5 15.6L17.81 6.63Z"
                  fill="currentColor"
                />
                <path
                  d="M4.81 6.63C4.48 6.37 4 6.61 4 7.02V24.98C4 25.4 4.48 25.63 4.81 25.37L15.5 16.39C15.76 16.19 15.76 15.8 15.5 15.6L4.81 6.63Z"
                  fill="currentColor"
                />
              </svg>
            </span>
          </button>
          <button
            aria-label="Go to last page"
            class="btn pagination-button"
          >
            <span
              aria-hidden="true"
            >
              <svg
                class="pagination-icon step-forward"
                fill="currentColor"
                viewBox="0 0 32 32"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8.81 6.63C8.48 6.37 8 6.61 8 7.02V24.98C8 25.4 8.48 25.63 8.81 25.37L20.5 16.39C20.76 16.19 20.76 15.8 20.5 15.6L8.81 6.63Z"
                  fill="currentColor"
                />
                <path
                  d="M24 6H22V26H24V6Z"
                  fill="currentColor"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
    </div>
    <section
      aria-label="Mock props header aria label"
      class="widget-gallery-header widget-gallery-filter"
    >
      <input />
    </section>
    <div
      class="widget-gallery-content"
    >
      <div
        aria-label="Mock props ListBox aria label"
        class="widget-gallery-items widget-gallery-lg-4 widget-gallery-md-3 widget-gallery-sm-2"
        role="list"
      >
        <div
          class="widget-gallery-item item-class"
          data-position="0,0"
          role="listitem"
          tabindex="0"
        >
          Item content
        </div>
        <div
          class="widget-gallery-item item-class"
          data-position="1,0"
          role="listitem"
          tabindex="-1"
        >
          Item content
        </div>
        <div
          class="widget-gallery-item item-class"
          data-position="2,0"
          role="listitem"
          tabindex="-1"
        >
          Item content
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Gallery without filters renders structure without header container 1`] = `
<DocumentFragment>
  <div
    class="widget-gallery my-gallery"
    data-focusindex="0"
  >
    <div
      class="widget-gallery-content infinite-loading"
    >
      <div
        aria-label="Mock props ListBox aria label"
        class="widget-gallery-items widget-gallery-lg-4 widget-gallery-md-3 widget-gallery-sm-2"
        role="list"
      >
        <div
          class="widget-gallery-item item-class"
          data-position="0,0"
          role="listitem"
          tabindex="0"
        >
          Item content
        </div>
        <div
          class="widget-gallery-item item-class"
          data-position="1,0"
          role="listitem"
          tabindex="-1"
        >
          Item content
        </div>
        <div
          class="widget-gallery-item item-class"
          data-position="2,0"
          role="listitem"
          tabindex="-1"
        >
          Item content
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
